docs/DeadmanAlert.md

```diff
@@ -0,0 +1,99 @@
+### Deadman Alert
+
+The `deadman` function is used in the context of a task to alert when a group stops reporting data.
+For example, say you have the following task definition:
+
+```
+option task = {
+    ...
+    every: 15m,
+    ...
+}
+from(bucket: "telegraf/autogen")
+    |> range(start: -1h)
+    |> filter(fn: (r) => r._measurement == "cpu")
+```
+
+In order to be alerted when the above task stops reporting data for a group, you would define a deadman alert like so:
+```
+option task = {
+    ...
+    every: 15m,
+    ...
+}
+f = (start, stop) =>
+    from(bucket: "telegraf/autogen")
+        |> range(start: start, stop: stop)
+        |> filter(fn: (r) => r._measurement == "cpu")
+
+deadman(f: f, period: 1h, offset: task.every)
+```
+
+where the deadman function is defined as follows:
+```
+deadman = (f, period, offset) => {
+
+    r = f(start: now() - offset - period, stop: now() - offset)
+        |> group_keys()
+
+    s = f(start: now() - period, stop: now())
+        |> group_keys()
+
+    return diff(r: r, s: s) |> alert(crit: (r) => true)
+}
+```
+
+Note `group_keys` takes in a stream of tables and returns the group keys of those tables in the output:
+```
+group_keys = (tables=<-) =>
+    |> keys()
+    |> limit(n:1)
+    |> drop(columns: ["_value"])
```

Comment:
Should this function end with `group()` to put all rows into a single table? Otherwise the diff computation below is per table and it will have to cache all tables in order to compute the diff.

Reply:
As each table will only have a single row, my assumption is that the overhead incurred by keeping each series key in its own table will be minimal. As cardinality increases that assumption might become tenuous. However by not performing a destructive operation like `group()`, we ensure the `deadman` function works for all heterogeneous data.

This of course assumes that `diff` or left anti-join works on heterogeneous data.